lms/djangoapps/commerce/tests.py

```diff
@@ -245,3 +245,15 @@ def test_no_settings(self):
         # Ensure that the user is not enrolled and that no calls were made to the E-Commerce API
         self.assertTrue(CourseEnrollment.is_enrolled(self.user, self.course.id))
         self.assertIsInstance(httpretty.last_request(), HTTPrettyRequestEmpty)
+
+    def test_existing_enrollment(self):
+        """ The view should respond with HTTP 200 if the user is already enrolled in the course. """
+
+        # Enroll user in the course
+        CourseEnrollment.enroll(self.user, self.course.id)
+        self.assertTrue(CourseEnrollment.is_enrolled(self.user, self.course.id))
+
+        response = self._post_to_view()
+        self.assertEqual(response.status_code, 200)
```

Comment:
The use of DRF is an implementation detail that I would like to keep out of the tests.